aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2695,14 +2695,32 @@ static void apply_gels(const Tensor& a, Tensor& b, Tensor& infos) {
 #endif
 }
 
+void gels_magma(const Tensor& a, Tensor& b, Tensor& infos) {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(a.scalar_type(), "gels_magma", [&] {
+    apply_gels<scalar_t>(a, b, infos);
```

Comment:
MAGMA code is not shown in this PR. Does that support broadcasting as well?

Reply:
It does, here are the tests https://github.com/pytorch/pytorch/blob/f84f2063b43a32fbe88c7e4de7e1019a00adb302/test/test_linalg.py#L250

It uses @nikitaved's `batch_iterator_with_broadcasting` and it's easy there as MAGMA calls are device synchronizing.